framework/tester.py

```diff
@@ -185,27 +240,38 @@ def setUp(self):
         time.sleep(0.2)
 
     def tearDown(self):
-        tf_cfg.dbg(3, "\tTeardown")
-        for cid in self.__clients:
-            client = self.__clients[cid]
-            client.stop()
-        self.__tempesta.stop()
-        for sid in self.__servers:
-            server = self.__servers[sid]
-            server.stop()
-        self.deproxy_manager.stop()
+        has_base_excpt = False
         try:
-            deproxy_manager.finish_all_deproxy()
-        except:
-            print ('Unknown exception in stopping deproxy')
-
-        self.oops.update()
-        if self.oops.warn_count("Oops") > 0:
-            raise Exception("Oopses happened during test on Tempesta")
-        if self.oops.warn_count("WARNING") > 0:
-            raise Exception("Warnings happened during test on Tempesta")
-        if self.oops.warn_count("ERROR") > 0:
-            raise Exception("Errors happened during test on Tempesta")
+            tf_cfg.dbg(3, "\tTeardown")
+            for cid in self.__clients:
+                client = self.__clients[cid]
+                client.stop()
+            self.__tempesta.stop()
+            for sid in self.__servers:
+                server = self.__servers[sid]
+                server.stop()
+            self.deproxy_manager.stop()
+            try:
+                deproxy_manager.finish_all_deproxy()
+            except:
+                print ('Unknown exception in stopping deproxy')
+
+            self.oops.update()
+            if self.oops.warn_count("Oops") > 0:
+                raise Exception("Oopses happened during test on Tempesta")
+            if self.oops.warn_count("WARNING") > 0:
+                raise Exception("Warnings happened during test on Tempesta")
+            if self.oops.warn_count("ERROR") > 0:
+                raise Exception("Errors happened during test on Tempesta")
+        except Exception as exc:
+            has_base_excpt = True
+            excpt = exc
+        tf_cfg.dbg(2, "Removing interfaces")
+        interface = tf_cfg.cfg.get('Server', 'aliases_interface')
+        remove_interfaces(interface, self.__ips)
```

Comment:
`remove_interfaces()` doesn't throw uncatched exceptions. It's better to  place `remove_interfaces()` call right between `self.oops.update()` and `if self.oops.warn_count....` that wrap everything with another try-catch block.

Reply:
Then if an exception is thrown in the block try, the interfaces will not be deleted.